### PR TITLE
[DOCS] Adds p-value heuristic to significant terms aggregation

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds the p-value scoring heuristic to the parameters of the significant terms aggregation. The section qualifies the p-value as a function provided by the ML plugin. It also contains an example.

Do not merge this PR before https://github.com/elastic/elasticsearch/pull/75313. Snippet test fails until that PR is merged.

### Preview

[p-value score](https://elasticsearch_75369.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-aggregations-bucket-significantterms-aggregation.html#_p_value_score)

### CODE REVIEW

File: docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc [hunk 1]
```diff
@@ -385,6 +385,98 @@ Google normalized distance as described in https://arxiv.org/pdf/cs/0412098v3.pd
 // NOTCONSOLE
 `gnd` also accepts the `background_is_superset` parameter.
 
+[p-value-score]
+===== p-value score
+
+[NOTE]
+====
+As p-value score heuristic is provided by the {ml} plugin. To use it, you must
```
Comment (lcawl) [inline docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc:393]: Perhaps the first sentence is unnecessary?
```suggestion
To use this heuristic, you must
```

File: docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc [hunk 2]
```diff
@@ -385,6 +385,98 @@ Google normalized distance as described in https://arxiv.org/pdf/cs/0412098v3.pd
 // NOTCONSOLE
 `gnd` also accepts the `background_is_superset` parameter.
 
+[p-value-score]
+===== p-value score
+
+[NOTE]
+====
+As p-value score heuristic is provided by the {ml} plugin. To use it, you must 
+have the {subscriptions}[appropriate subscription].
```
Comment (lcawl) [inline docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc:394]:
```suggestion
have the {subscriptions}[appropriate subscription] for {ml-features}.
```

File: docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc [hunk 3]
```diff
@@ -385,6 +385,98 @@ Google normalized distance as described in https://arxiv.org/pdf/cs/0412098v3.pd
 // NOTCONSOLE
 `gnd` also accepts the `background_is_superset` parameter.
 
+[p-value-score]
```
Comment (szabosteve) [inline docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc:388]:
```suggestion
[[p-value-score]]
```

**Comment by szabosteve** [comment]
@elasticmachine update branch

File: docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc [hunk 4]
```diff
@@ -385,6 +385,100 @@ Google normalized distance as described in https://arxiv.org/pdf/cs/0412098v3.pd
...
 `gnd` also accepts the `background_is_superset` parameter.
 
+[role="xpack"]
+[[p-value-score]]
+===== p-value score
+
+[NOTE]
+====
+To use this heuristic, you must
+have the {subscriptions}[appropriate subscription] for {ml-features}.
```
Comment (benwtrent) [inline docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc:395]: I am not sure how to best word this. It's "basic". So, it is free if the ML plugin is enabled. 

So, they just need to have the ML plugin enabled. I am not sure about "subscription"

**Review by benwtrent — COMMENTED** [review]
one minor thing. The rest looks good.

File: docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc (hunk 4 above)
Comment (lcawl) [reply docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc:395]: Can you clarify how to enable the plugin? We currently only describe how to add ML nodes, but this seems like something else.

Comment (benwtrent) [reply docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc:395]: @lcawl , it just means that the plugin is part of the distribution. There is a way to disable the plugin by setting `xpack.ml.enabled` to `false`. You are talking about nodes that are valid places to run ML jobs (the node role setting).

Comment (lcawl) [reply docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc:395]: In this case, my suggestion is that we remove this "note" but leave the role=xpack and make sure the p-score functionality is added to the subscriptions page in the appropriate subscription tier.

Comment (szabosteve) [reply docs/reference/aggregations/bucket/significantterms-aggregation.asciidoc]: I removed the note via f7255d5

**Comment by szabosteve** [comment]
@elasticmachine merge upstream

**Review by lcawl — APPROVED** [review]

**Pull request merged**